Dockerfile

```diff
@@ -1,9 +1,10 @@
 # Perform common operations, dependency installation etc...
-FROM python:3.9.0-buster as base
+FROM python:3.8-slim-buster as base
 RUN pip install poetry
 WORKDIR /DevOps-Course-Starter
 COPY pyproject.toml /DevOps-Course-Starter/
 COPY . /DevOps-Course-Starter/
+RUN poetry config virtualenvs.create false && poetry install --no-interaction
```

Comment:
You shouldn't have `poetry install` both here and in the `production` stage. 
I would recommend you remove it from here so that you can have `poetry install --no-dev` in the production stage and update the test stage to do `poetry install` without the no-dev flag.